packages/components/src/Toast/Toast.tsx

```diff
@@ -54,24 +63,94 @@ const ToastAlert = styled(Alert)`
   }
 `
 
+const SpinnerContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 48px;
+  width: 48px;
+  margin-right: -16px;
+`
+
+const ActionLink = styled(Link)`
+  height: 24px;
+  margin-top: 12px;
+  margin-right: 4px;
+`
+
+const Close = styled(Button)`
+  color: ${({ theme }) => theme.colors.white};
+  margin-top: 4px;
+  margin-right: 4px;
+  &:hover {
+    background: var(--color) !important;
+  }
+`
+
+const NotificationMessage = styled.div`
+  ${({ theme }) => theme.fonts.bold16};
+  color: ${({ theme }) => theme.colors.white};
+  position: relative;
+  width: 100%;
+  padding: 14px 24px 12px 16px;
+  flex: 1;
+`
+
+export interface IToastProps extends React.HTMLAttributes<HTMLDivElement> {
+  type: ToastType
+  onClose?: (event?: React.MouseEvent<HTMLButtonElement>) => void
+  onActionClick?: (event?: React.MouseEvent<HTMLButtonElement>) => void
+  actionText?: string
+  duration?: number | null
+}
+
 export const Toast = ({
   type,
-  duration = TOAST_DEFAULT_DURATION_MS,
   onClose,
+  duration = TOAST_DEFAULT_DURATION_MS,
+  onActionClick,
+  actionText,
+  children,
   ...props
-}: ToastProps) => {
-  useToastVisibility({
-    duration: type === 'loading' ? null : duration,
-    onClose
-  })
+}: IToastProps) => (
+  <Container $type={type} {...props}>
+    {type === 'loading' && (
+      <SpinnerContainer>
+        <Spinner
+          id="in-progress-floating-notification"
+          baseColor={colors.white}
+          size={20}
+        />
+      </SpinnerContainer>
+    )}
 
-  return (
-    <ToastAlert
-      type={type ?? 'error'}
-      onClose={onClose}
-      role="alert"
-      aria-live="polite"
-      {...props}
-    />
-  )
-}
+    <NotificationMessage>{children}</NotificationMessage>
+
+    {useToastVisibility({
+      duration: type === 'loading' ? null : duration,
+      onClose
+    })}
```

Comment:
This might work here, but for clarity it should rather not be with the components in the return clause. The more common way is something like
```
export const Toast = (..) => {
  useToastVisibility(..);
  
  return <>Your JSX components here...</>
}
```